.github/workflows/integration.yaml

```diff
@@ -166,32 +134,225 @@ jobs:
+            config-state: "skip"
+          - consent: "skip"
+            consent-registry-flag: "0"
+            config-state: "null"
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v6
+
+      - name: Download artifacts
+        uses: actions/download-artifact@v5
+        with:
+          path: ci-artifacts
+
+      - name: Get WSL image
+        uses: ./.github/actions/get-wsl-image
+        with:
+          output_path: images.wsl
+
+      - name: Prepare Ubuntu Insights consent registry key
+        if: matrix.consent-registry-flag != 'skip'
+        shell: powershell
+        run: |
+          New-Item -Path "${{ env.insights_registry_key_path }}" -Force
+          New-ItemProperty -Path "${{ env.insights_registry_key_path }}" -Name "${{ env.insights_registry_value_name }}" -Value "${{ matrix.consent-registry-flag }}" -PropertyType DWord -Force
+
+      - name: Install WSL instance
+        shell: powershell
+        env:
+          WSL_UTF8: "1"
+        run: |
+          wsl --install --no-launch --from-file "images.wsl" --name "${{ env.instance }}"
+
+      - name: Install wsl-setup and test dependencies
+        shell: powershell
+        env:
+          WSL_UTF8: "1"
+        run: |
+          # Install the deb package
+          wsl -d "${{ env.instance }}" -u root -- dpkg -i "./ci-artifacts/wsl-setup_*/wsl-setup_*.deb"
+          # Install expect
+          wsl -d "${{ env.instance }}" -u root -- apt-get update
+          wsl -d "${{ env.instance }}" -u root -- apt-get install -y expect
+
+      - name: Configure consent file
+        if: matrix.config-state != 'skip'
+        shell: powershell
+        env:
+          WSL_UTF8: "1"
+        run: |
+          $configContent = "consent_state = ${{ matrix.config-state }}"
+          wsl -d "${{ env.instance }}" -u root -- bash -c "mkdir -p /etc/skel/.config/ubuntu-insights"
+          wsl -d "${{ env.instance }}" -u root -- bash -c "echo '$configContent' > /etc/skel/.config/ubuntu-insights/consent.toml"
+
+      - name: Reset WSL instance state
+        shell: powershell
+        env:
+          WSL_UTF8: "1"
+        run: |
+          wsl -d "${{ env.instance }}" -u root -- cloud-init status --wait
+          wsl -d "${{ env.instance }}" -u root -- bash -ec "rm /etc/cloud/cloud-init.disabled || true"
+          wsl -d "${{ env.instance }}" -u root -- cloud-init clean --logs
+          wsl --shutdown
+
+      - name: Run expect test
+        shell: powershell
+        env:
+          WSL_UTF8: "1"
+        timeout-minutes: 10
+        run: |
+          # Copy expect script to instance
+          Get-Content "test/e2e/setup.expect" -Encoding UTF8 | wsl -d "${{ env.instance }}" -u root -- bash -c "cat > /tmp/setup.expect"
+
+          # Run the expect script
+          wsl -d "${{ env.instance }}" -u root -- expect /tmp/setup.expect "${{ env.username }}" "${{ env.password }}" "${{ matrix.consent }}"
+
+      - name: Determine expected consent state
```

Comment:
I'm slightly puzzled about the need for this translation step. Couldn't we have the input already formatted in the shape the test expects?